### PR TITLE
compatibility changes for dadbattle + philly

### PR DESCRIPTION
**Pull request by Lucky-56**
fixes:

- 'Dad Battle' now has compatibility again *yay*

'Dad Battle' is now called 'Dad Battle' in game and uses 'dadbattle' as files
'Philly' is now called 'Philly Nice' in game and uses 'philly' as files (reverted #760 and instead fixed it better)

### CODE REVIEW

**Comment by Lucky-56** [comment]
@theDetourist here ya go

File: source/FreeplayState.hx [hunk 1]
```diff
@@ -202,13 +202,24 @@ class FreeplayState extends MusicBeatState
 
 		if (accepted)
 		{
-			trace(StringTools.replace(songs[curSelected].songName," ", "-").toLowerCase());
+			// pre lowercasing the song name
+			var songLowercase = switch (songs[curSelected].songName)
```
Comment (puyoxyz) [inline source/FreeplayState.hx]: This switch case is hella wack you're repeating code and changing like one argument

Do something like this instead

```haxe
var songLowercase = StringTools.replace(PlayState.SONG.song, " ", "-").toLowerCase();
switch (songLowercase) {
  case 'dad-battle': songLowercase = 'dadbattle';
  case 'philly-nice': songLowercase = 'philly';
}
```

File: source/Paths.hx [hunk 2]
```diff
@@ -97,14 +97,30 @@ class Paths
 
 	inline static public function voices(song:String)
 	{
-		song = StringTools.replace(song," ", "-");
-		return 'songs:assets/songs/${song.toLowerCase()}/Voices.$SOUND_EXT';
+		var songLowercase = switch (song)
```
Comment (puyoxyz) [inline source/Paths.hx]: See previous

File: source/Paths.hx [hunk 3]
```diff
@@ -97,14 +97,30 @@ class Paths
...
+					StringTools.replace(song," ", "-").toLowerCase();
+			}
+		return 'songs:assets/songs/${songLowercase}/Voices.$SOUND_EXT';
 	}
 
 	inline static public function inst(song:String)
 	{
-		song = StringTools.replace(song," ", "-");
-		return 'songs:assets/songs/${song.toLowerCase()}/Inst.$SOUND_EXT';
+		var songLowercase = switch (song)
```
Comment (puyoxyz) [inline source/Paths.hx]: see previous

File: source/PlayState.hx [hunk 4]
```diff
@@ -236,14 +236,25 @@ class PlayState extends MusicBeatState
 		repPresses = 0;
 		repReleases = 0;
 
+		// pre lowercasing the song name (create)
```
Comment (puyoxyz) [inline source/PlayState.hx:239]: see previous

File: source/PlayState.hx [hunk 5]
```diff
@@ -1071,11 +1082,21 @@ class PlayState extends MusicBeatState
 		senpaiEvil.updateHitbox();
 		senpaiEvil.screenCenter();
 
-		if (SONG.song.toLowerCase() == 'roses' || SONG.song.toLowerCase() == 'thorns')
+		// pre lowercasing the song name (schoolIntro)
```
Comment (puyoxyz) [inline source/PlayState.hx:1081]: see previous

File: source/PlayState.hx [hunk 6]
```diff
@@ -1368,9 +1389,20 @@ class PlayState extends MusicBeatState
 
 		var playerCounter:Int = 0;
 
+		// pre lowercasing the song name (generateSong)
+		var songLowercase = switch (PlayState.SONG.song)
```
Comment (puyoxyz) [inline source/PlayState.hx]: see previous

File: source/PlayState.hx [hunk 7]
```diff
@@ -2498,9 +2530,29 @@ class PlayState extends MusicBeatState
 						difficulty = '-hard';
 
 					trace('LOADING NEXT SONG');
-					trace(PlayState.storyPlaylist[0].toLowerCase() + difficulty);
+					// pre lowercasing the next story song name
```
Comment (puyoxyz) [inline source/PlayState.hx:2529]: see previous

File: source/PlayState.hx [hunk 8]
```diff
@@ -2498,9 +2530,29 @@ class PlayState extends MusicBeatState
...
+						case 'Philly Nice':
+							StringTools.replace(PlayState.storyPlaylist[0]," Nice", "").toLowerCase();
+						default:
+							StringTools.replace(PlayState.storyPlaylist[0]," ", "-").toLowerCase();
+					}
+					trace(nextSongLowercase + difficulty);
 
-					if (SONG.song.toLowerCase() == 'eggnog')
+					// pre lowercasing the song name (endSong)
+					var songLowercase = switch (PlayState.SONG.song)
```
Comment (puyoxyz) [inline source/PlayState.hx]: see previous

File: source/Song.hx [hunk 9]
```diff
@@ -46,9 +46,23 @@ class Song
 
 	public static function loadFromJson(jsonInput:String, ?folder:String):SwagSong
 	{
-		trace('loading ' + StringTools.replace(folder," ", "-").toLowerCase() + '/' + StringTools.replace(jsonInput," ", "-").toLowerCase());
-
-		var rawJson = Assets.getText(Paths.json(StringTools.replace(folder," ", "-").toLowerCase() + '/' + StringTools.replace(jsonInput," ", "-").toLowerCase())).trim();
+		trace(jsonInput);
+		// pre lowercasing the folder name
+		var folderLowercase = switch (folder)
```
Comment (puyoxyz) [inline source/Song.hx]: see previous

**Review by puyoxyz — CHANGES_REQUESTED** [review]

**Comment by Lucky-56** [comment]
sure, changes comin' right up

**Review by puyoxyz — APPROVED** [review]
Looks good; @Lucky-56 did you test it with the new changes?

**Comment by Lucky-56** [comment]
ok... wierd things are happening.
I start Philly in freeplay, but I play Satin Panties... ?

**Comment by Lucky-56** [comment]
ah ok I broke Freeplay, nothing I can't fix

**Comment by Lucky-56** [comment]
it just crashes when going into freeplay now.. and I'm too tired to fix it now :/

**Comment by Lucky-56** [comment]
it works now tested it completely

File: source/FreeplayState.hx [hunk 10]
```diff
@@ -202,13 +202,27 @@ class FreeplayState extends MusicBeatState
...
+				case 'dad-battle': songLowercase = 'dadbattle';
+				case 'philly-nice': songLowercase = 'philly';
+			}
+			// adjusting the highscore song name to be compatible (update)
+			// would read original scores if we didn't change packages
+			var songHighscore = StringTools.replace(songs[curSelected].songName, " ", "-");
+			switch (songHighscore) {
+				case 'Dad-Battle': songHighscore = 'Dadbattle';
+				case 'Philly-Nice': songHighscore = 'Philly';
+			}
```
Comment (puyoxyz) [inline source/FreeplayState.hx:217]: there *has to* be a less janky way to do this but this whole codebase is wack so whatever

**Review by puyoxyz — APPROVED** [review]
Yep

**Comment by Lucky-56** [comment]
just checked and remembered
1. it doesn't even get called at the beginning of the song
2. it's only a problem on my mod  😅

**Review by isakube — APPROVED** [review]
imagine making a commit and then immediately reverting it LOLL

**Comment by Lucky-56** [comment]
yeah because I noticed it right when I commited... heh

**Comment by Lucky-56** [comment]
Fixes #779 
#790 don't know yet would assume it fixes that
Fixes #796 
Fixes #804 
Fixes #805  
i guess

**Review by RagdollClash — CHANGES_REQUESTED** [review]
Satin Panties and Winter Horrorland also need compatibility changes.

**Review by RagdollClash — APPROVED** [review]
nvm, disregard what i said

**Comment by puyoxyz** [comment]
@KadeDev ??????????????????

**Comment by Kade-github** [comment]
sorryy!!!

**Comment by Lucky-56** [comment]
yeah it's only hardcoded because it's for compatibility

**Pull request merged**